src/Series.cpp

```diff
@@ -350,6 +350,16 @@ Series::setName(std::string const& n)
     return *this;
 }
 
+void Series::setFlush(FlushType flushType)
+{
+    IOHandler->flushType = flushType;
```

Comment:
To solve the redundant loop, we could `Series::flush()` when switching from `DEFER` TO `DIRECT`. This would also schedule and perform all chunk transfers that would be done in the loops.

Reply:
I think of the look as a safety net at negligible costs.
Nevertheless, we can start flushing around here as well, that makes sense.